theano/sandbox/cuda/opt.py

```diff
@@ -2313,3 +2313,73 @@ def local_gpu_allocempty(node):
                'scan')
 
 import theano.sandbox.cuda.extra_ops
+import theano.compile.builders
+
+
+@register_opt()
+@local_optimizer([theano.compile.builders.OpFromGraph])
+def local_gpu_op_from_graph(node):
+    if (isinstance(node.op, theano.compile.builders.OpFromGraph) and
+        any([(i.owner and isinstance(i.owner.op, HostFromGpu))
+             for i in node.inputs]) and
+        # If the output is on the GPU, we already moved it to the GPU.
+        not any([isinstance(o.type, CudaNdarrayType) for o in node.outputs])):
+            # Need to implement updates, givens, ...
+            if node.op.kwargs:
+                for k in node.op.kwargs.keys():
+                    if k not in ['mode']:
+                        return
```

Comment:
Currently, the opt only know how to handle the mode parameter. So this is incomplete. I didn't taught what to do with the other param.